code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm

```diff
@@ -54,21 +55,14 @@
 	var/datum/antagonist/heretic/master = user.mind.has_antag_datum(/datum/antagonist/heretic)
 	heretic_monster.set_owner(master)
 	atoms -= humie
+	RegisterSignal(humie,COMSIG_MOB_DEATH,.proc/remove_ghoul)
 	ghouls += humie
 
-/datum/eldritch_knowledge/flesh_ghoul/proc/check_ghouls(mob/living/user)
-
-	listclearnulls(ghouls)
-
-	for(var/mob/living/carbon/human/ghoul in ghouls)
-		if(ghoul.stat == DEAD)
-			ghoul?.mind.remove_antag_datum(/datum/antagonist/heretic_monster)
-			ghouls -= ghoul
-			current_amt--
-
-	if(current_amt >= max_amt)
-		return FALSE
-	return TRUE
+/datum/eldritch_knowledge/flesh_ghoul/proc/remove_ghoul(datum/source)
+	var/mob/living/carbon/human/humie = source
+	ghouls -= humie
+	humie?.mind.remove_antag_datum(/datum/antagonist/heretic_monster)
```

Comment:
Why are you checking if the `humie` exists here? It must, else there would be no signal triggered.

Reply:
good catch